src/API.jl

```diff
@@ -724,7 +725,10 @@ function instantiate(ctx::Context; manifest::Union{Bool, Nothing}=nothing,
                  " Otherwise, remove `$name` with `Pkg.rm(\"$name\")`.",
                  " Finally, run `Pkg.instantiate()` again.")
     end
+    # Download artifacts for this here package before fast-exiting
+    Operations.download_artifacts(ctx, [dirname(ctx.env.manifest_file)]; platform=platform, verbose=verbose)
     Operations.is_instantiated(ctx) && return
```

Comment:
`is_instantiated` probably needs to be updated too, right?

Reply:
I guess it depends on what you want it to do; would we want it to return `false` if any artifacts in any dependencies are not instantiated?  That kind of changes the meaning a bit, but would make sense to me.

Reply:
Yea, otherwise we exit here without functioning packages.